src/query-rule-context/query-rule-context.ts

```diff
@@ -0,0 +1,34 @@
+import { Component, Input, Inject, forwardRef } from '@angular/core';
+
+import { connectQueryRules } from 'instantsearch.js/es/connectors';
+import { BaseWidget } from '../base-widget';
+import { NgAisInstantSearch } from '../instantsearch/instantsearch';
+
+type FacetValue = string | number | boolean;
+
+@Component({
+  selector: 'ais-query-rule-context',
+  template: '',
+})
+export class NgAisQueryRuleContext extends BaseWidget {
+  @Input() public trackedFilters: {
+    [facetName: string]: (facetValues?: FacetValue[]) => FacetValue[];
```

Comment:
Facet values are always passed by the connector. If there are none, an empty array is passed.

```suggestion
    [facetName: string]: (facetValues: FacetValue[]) => FacetValue[];
```